samples/gibbs_ensemble_socket.py

```diff
@@ -0,0 +1,437 @@
+    boxes[(rand_box + 1) % 2].send_data(pickle.dumps([MSG_CHANGE_VOLUME,
+                                                      boxes[(rand_box + 1) % 2].box_l]))
+
+    boxes[rand_box].recv_energy()
+    boxes[(rand_box + 1) % 2].recv_energy()
+    return rand_box
+
+
+def exchange_particle(boxes):
+    '''
+    Tries a exchange exchange move and stores the new energy in the boxes
+    '''
+    rand_box = random.randint(0, 1)
+    if(boxes[rand_box].n_particles == 0):
+        rand_box = (rand_box + 1) % 2
+    boxes[rand_box].n_particles -= 1
+    boxes[(rand_box + 1) % 2].n_particles += 1
+    boxes[rand_box].send_data(pickle.dumps([MSG_EXCHANGE_PART_REMOVE, 0]))
+    boxes[(rand_box + 1) %
+          2].send_data(pickle.dumps([MSG_EXCHANGE_PART_ADD, 0]))
+
+    boxes[rand_box].recv_energy()
+    boxes[(rand_box + 1) % 2].recv_energy()
+    return rand_box
+
+
+def check_make_move(boxes, inner_potential, rand_box):
+    '''
+    Returns whether the last displacement move was valid or not and reverts the changes
+    if it was invalid.
+    '''
+    if(random.random() > inner_potential):
+        boxes[rand_box].send_data(pickle.dumps([MSG_MOVE_PART_REVERT, 0]))
+        boxes[rand_box].recv_energy()
+        return False
+    return True
+
+
+def check_exchange_volume(boxes, inner_potential):
+    '''
+    Returns whether the last volume exchange move was valid or not and reverts the changes
+    if it was invalid.
+    '''
+    volume_factor = \
+        (boxes[0].box_l**3 / boxes[0].box_l_old**3)**(boxes[0].n_particles + 1) * \
+        (boxes[1].box_l**3 / boxes[1].box_l_old **
+         3)**(boxes[1].n_particles + 1)
+    if random.random() > volume_factor * inner_potential:
+        boxes[0].send_data(pickle.dumps(
+            [MSG_CHANGE_VOLUME, boxes[0].box_l_old]))
+        boxes[0].box_l = boxes[0].box_l_old
+        boxes[1].send_data(pickle.dumps(
+            [MSG_CHANGE_VOLUME, boxes[1].box_l_old]))
+        boxes[1].box_l = boxes[1].box_l_old
+
+        boxes[0].recv_energy()
+        boxes[1].recv_energy()
+        return False
+    return True
+
+
+def check_exchange_particle(boxes, inner_potential, rand_box):
+    '''
+    Returns whether the last particle exchange move was valid or not and reverts the changes
+    if it was invalid.
+    '''
+    exchange_factor = (boxes[rand_box].n_particles) / \
+        (boxes[(rand_box + 1) % 2].n_particles + 1.0) * \
+        boxes[(rand_box + 1) % 2].box_l**3 / boxes[rand_box].box_l**3
+    if random.random() > exchange_factor * inner_potential:
+        boxes[rand_box].n_particles += 1
+        boxes[(rand_box + 1) % 2].n_particles -= 1
+        boxes[rand_box].send_data(pickle.dumps(
+            [MSG_EXCHANGE_PART_REMOVE_REVERT, 0]))
+        boxes[(rand_box + 1) %
+              2].send_data(pickle.dumps([MSG_EXCHANGE_PART_ADD_REVERT, 0]))
+
+        boxes[rand_box].recv_energy()
+        boxes[(rand_box + 1) % 2].recv_energy()
+        return False
+    return True
+
+
+
```

Comment:
it would probably be helpful to guard the module-level code by ```if __name__ == "__main__":```. This makes it easier to reuse the functionality in this module